Doc/c-api/datetime.rst

```diff
@@ -79,27 +89,41 @@ Macros to create objects:
 
 .. c:function:: PyObject* PyDate_FromDate(int year, int month, int day)
 
-   Return a ``datetime.date`` object with the specified year, month and day.
+   Return a :class:`datetime.date` object with the specified year, month and day.
 
 
 .. c:function:: PyObject* PyDateTime_FromDateAndTime(int year, int month, int day, int hour, int minute, int second, int usecond)
 
-   Return a ``datetime.datetime`` object with the specified year, month, day, hour,
+   Return a :class:`datetime.datetime` object with the specified year, month, day, hour,
    minute, second and microsecond.
 
 
 .. c:function:: PyObject* PyTime_FromTime(int hour, int minute, int second, int usecond)
 
-   Return a ``datetime.time`` object with the specified hour, minute, second and
+   Return a :class:`datetime.time` object with the specified hour, minute, second and
    microsecond.
 
 
 .. c:function:: PyObject* PyDelta_FromDSU(int days, int seconds, int useconds)
 
-   Return a ``datetime.timedelta`` object representing the given number of days,
-   seconds and microseconds.  Normalization is performed so that the resulting
-   number of microseconds and seconds lie in the ranges documented for
-   ``datetime.timedelta`` objects.
+   Return a :class:`datetime.timedelta` object representing the given number
+   of days, seconds and microseconds.  Normalization is performed so that the
+   resulting number of microseconds and seconds lie in the ranges documented for
+   :class:`datetime.timedelta` objects.
+
+.. c:function:: PyObject* PyTimeZone_FromOffset(PyDateTime_DeltaType* offset)
```

Comment:
When I build the documentation, I see *Return value: New reference.* under all the constructor macros *except* these two, but I'm not sure where that's being generated from. I don't see why they *wouldn't* be a new reference, but I don't 100% grok when something does and does not change the reference count.

Reply:
Reference counting information is stored in Doc/data/refcounts.dat

Reply:
@abalkin When documenting the reference counts, I ran into a bit of a problem, since the reference to `offset` is not increased if `offset` is equivalent to `timedelta(0)`, so it's ambiguous whether the function's effect on `offset`'s reference count is `+1` or `0`. I went with `+1` and added a comment to the effect that it's not always true.

Reply:
I am going to merge this now.  We can fine tune the documentation during the beta phase.